WinCairo build fix after 261225@main

dpino opened the pull request:
#### 89cc5666c25adffd7aff6795b63e0186b8f55ccc
<pre>
WinCairo build fix after 261225@main
<a href="https://bugs.webkit.org/show_bug.cgi?id=253348">https://bugs.webkit.org/show_bug.cgi?id=253348</a>

Reviewed by Chris Dumez.

Remove inline keyword in WebChromeClient dtor and set definition to default.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::~WebChromeClient): Deleted.

Canonical link: <a href="https://commits.webkit.org/261327@main">https://commits.webkit.org/261327@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/7425abd8377b5dbaa4dbd0bda7696c8eab9907ea

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/111298 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/20441 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/43843 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/87/builds/2728 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/12/builds/120106 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/21806 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/11540 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/86/builds/2353 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/117063 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/16197 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/99375 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/103930 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/98148 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/31027 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/35/builds/44782 "Found 1 new test failure: webgl/2.0.0/conformance2/textures/image_bitmap_from_video/tex-2d-r8ui-red_integer-unsigned_byte.html (failure)") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/12940 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/32368 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/86621 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/13457 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/84/builds/9359 "Passed tests") | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/18897 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/62/builds/51951 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/7851 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/15413 "Built successfully") | | | 
| | | | | 
<!--EWS-Status-Bubble-End-->

fujii commented on Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
```diff
@@ -166,10 +166,6 @@ void WebChromeClient::didRemoveMenuItemElement(HTMLMenuItemElement& element)
     m_page.didRemoveMenuItemElement(element);
 }
 
-inline WebChromeClient::~WebChromeClient()
```
Generally, non inlined dtor is a good practice to reduce the number of headers to include.
If you don't have a performance problem you have to inline, I'd like to keep it non inlined.

cdumez replied:
`WebChromeClient::~WebChromeClient() = default;` in the cpp sounds good to me.

dpino replied:
I don't fully understand. Originally the dtor was already inlined. Moving the definition of dtor to the header preserves the inline nature of the dtor, since method definitions in a header are always inlined.

My motiviation is not performance, it's simply to fix this link error I was getting on Windows: https://bugs.webkit.org/show_bug.cgi?id=253348#c4

cdumez replied:
I understand, and there are multiple ways to fix it. The destructor doesn't need to be inlined here.

fujii replied:
I think marking methods with `inline` in a cpp file has no actual effect. So, this dtor wasn't inlined, I think.

fujii replied:
https://github.com/WebKit/WebKit/commit/0df6f3c9e212ad674dcd048424981066c87d950a added the dtor in the cpp file.
I think it's the author's intention. Marking `inline` was just a mistake.

achristensen07 replied:
I like Chris's suggestion.  "inline" should be removed.

cdumez reviewed (approved):
LGTM

dpino committed:
WinCairo build fix after 261225@main
https://bugs.webkit.org/show_bug.cgi?id=253348

Reviewed by Chris Dumez.

Remove inline keyword in WebChromeClient dtor and set definition to default.

* Source/WebKit/WebProcess/WebCoreSupport/WebChromeClient.cpp:
(WebKit::WebChromeClient::~WebChromeClient): Deleted.

Canonical link: https://commits.webkit.org/261327@main
